src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1347,248 +1361,56 @@ bool OutputStateMachineEngine::DispatchIntermediatesFromEscape() const noexcept
-        }
-    }
-    // Only if we find a valid colorspec can we pass it out successfully.
-    if (foundValidColorSpec)
-    {
-        DWORD color = RGB(LOBYTE(colorValues.at(0)),
-                          LOBYTE(colorValues.at(1)),
-                          LOBYTE(colorValues.at(2)));
-
-        rgb = color;
-        success = true;
-    }
-    return success;
-}
-
 // Routine Description:
 // - OSC 4 ; c ; spec ST
 //      c: the index of the ansi color table
-//      spec: a color in the following format:
-//          "rgb:<red>/<green>/<blue>"
-//          where <color> is two hex digits
+//      spec: The colors are specified by name or RGB specification as per XParseColor
+//
+//   It's possible to have multiple "c ; spec" pairs, which will set the index "c" of the color table
+//   with color parsed from "spec" for each pair respectively.
 // Arguments:
 // - string - the Osc String to parse
-// - tableIndex - receives the table index
-// - rgb - receives the color that we parsed in the format: 0x00BBGGRR
+// - tableIndexes - receives the table indexes
+// - rgbs - receives the colors that we parsed in the format: 0x00BBGGRR
 // Return Value:
-// - True if a table index and color was parsed successfully. False otherwise.
+// - True if at least one table index and color was parsed successfully. False otherwise.
 bool OutputStateMachineEngine::_GetOscSetColorTable(const std::wstring_view string,
-                                                    size_t& tableIndex,
-                                                    DWORD& rgb) const noexcept
+                                                    std::vector<size_t>& tableIndexes,
+                                                    std::vector<DWORD>& rgbs) const noexcept
+try
 {
-    tableIndex = 0;
-    rgb = 0;
-    size_t _TableIndex = 0;
+    std::vector<size_t> newTableIndexes;
+    std::vector<DWORD> newRgbs;
 
-    bool foundTableIndex = false;
-    bool success = false;
-    // We can have anywhere between [11,16] characters
-    // 11 "#;rgb:h/h/h"
-    // 16 "###;rgb:hh/hh/hh"
-    // Any fewer cannot be valid, and any more will be too many.
-    // Return early in this case.
-    //      We'll still have to bounds check when parsing the hh/hh/hh values
-    if (string.size() < 11 || string.size() > 16)
+    const auto parts = Utils::SplitString(string, L';');
+    if (parts.size() <= 1)
     {
         return false;
     }
 
-    // First try to get the table index, a number between [0,256]
-    size_t current = 0;
-    for (size_t i = 0; i < 4; i++)
+    for (size_t i = 0, j = 1; j < parts.size(); i += 2, j += 2)
     {
-        const wchar_t wch = string.at(current);
-        if (_isNumber(wch))
-        {
-            _TableIndex *= 10;
-            _TableIndex += wch - L'0';
-
-            ++current;
-        }
-        else if (wch == L';' && i > 0)
+        unsigned int tableIndex = 0;
+        const bool indexSuccess = Utils::StringToUint(parts.at(i), tableIndex);

```

Comment:
I can not use `til::at` here (and other similar places) because auditing complains about `C26445`.

Reply:
Seems like a false positive: `til::at` should return a reference, so a reference to a `std::basic_string_view` is perfectly reasonable here, and the only reasonable return value. Perhaps suppress the check here for now?

From the outside, it seems that `at` should be supressing that warning on its return value, if that can be expressed to the static analyser. That's a larger change though.

Reply:
Is `stoul` more reasonable than a hand-written `StringToUint`? I like the simplicity of `stoul` but it throws exception when there's invalid argument.

Reply:
nah -- just use a pragma warning suppress on this line for C26445 and use til::at.

Reply:
Suppressing on this specific line does not work. Suppressing inside `til::at` works, but that's not really idea, isn't it.